plugins/linux-pipewire/pipewire.c

```diff
@@ -584,23 +593,46 @@ static enum video_range_type video_color_range_from_spa_color_range(enum spa_vid
 	}
 }
 
-static bool prepare_obs_frame(obs_pipewire_stream *obs_pw_stream, struct obs_source_frame *frame)
+static bool prepare_obs_frame(obs_pipewire_stream *obs_pw_stream, struct spa_buffer *buffer,
+			      struct obs_source_frame *frame)
 {
 	struct obs_pw_video_format obs_pw_video_format;
 
-	frame->width = obs_pw_stream->format.info.raw.size.width;
-	frame->height = obs_pw_stream->format.info.raw.size.height;
+	switch (obs_pw_stream->format.media_subtype) {
+	case SPA_MEDIA_SUBTYPE_raw:
+		frame->width = obs_pw_stream->format.info.raw.size.width;
+		frame->height = obs_pw_stream->format.info.raw.size.height;
 
-	video_format_get_parameters(video_colorspace_from_spa_color_matrix(obs_pw_stream->format.info.raw.color_matrix),
-				    video_color_range_from_spa_color_range(obs_pw_stream->format.info.raw.color_range),
-				    frame->color_matrix, frame->color_range_min, frame->color_range_max);
+		video_format_get_parameters(
+			video_colorspace_from_spa_color_matrix(obs_pw_stream->format.info.raw.color_matrix),
+			video_color_range_from_spa_color_range(obs_pw_stream->format.info.raw.color_range),
+			frame->color_matrix, frame->color_range_min, frame->color_range_max);
 
-	if (!obs_pw_video_format_from_spa_format(obs_pw_stream->format.info.raw.format, &obs_pw_video_format) ||
-	    obs_pw_video_format.video_format == VIDEO_FORMAT_NONE)
-		return false;
+		if (!obs_pw_video_format_from_spa_format(obs_pw_stream->format.info.raw.format, &obs_pw_video_format) ||
+		    obs_pw_video_format.video_format == VIDEO_FORMAT_NONE)
+			return false;
+
+		frame->format = obs_pw_video_format.video_format;
+		frame->linesize[0] = SPA_ROUND_UP_N(frame->width * obs_pw_video_format.bpp, 4);
+
+		for (uint32_t i = 0; i < buffer->n_datas && i < MAX_AV_PLANES; i++) {
+			frame->data[i] = buffer->datas[i].data;
+			if (frame->data[i] == NULL) {
+				blog(LOG_ERROR, "[pipewire] Failed to access data");
+				return false;
+			}
+		}
+
+		break;
+	default:
```

Comment:
```suggestion
	case SPA_MEDIA_SUBTYPE_mjpg:
	case SPA_MEDIA_SUBTYPE_h264:
```

Note that a `default` still needs to be added but it should be  a `return false`.